src/xunit.analyzers/AssertIsTypeShouldUseGenericTypeFixer.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Xunit.Analyzers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public class AssertIsTypeShouldUseGenericTypeFixer : CodeFixProvider
+    {
+        const string titleTemplate = "Use Assert<{0}>.{1}";
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(Constants.Descriptors.X2007_AssertIsTypeShouldUseGenericType.Id);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var syntaxNode = root.FindNode(context.Span);
+            var invocation = syntaxNode.FirstAncestorOrSelf<InvocationExpressionSyntax>();
+            
+            var methodName = context.Diagnostics[0].Properties[AssertIsTypeShouldUseGenericType.MethodName];
+            var typeName = context.Diagnostics[0].Properties[AssertIsTypeShouldUseGenericType.TypeName];
+            var title = String.Format(titleTemplate, typeName, methodName);
+            
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    title,
+                    createChangedDocument: ct => RemoveTypeofInvocationAndAddGenericTypeAsync(context.Document, invocation, ct),
+                    equivalenceKey: title),
```

Comment:
Consider a less specific equivalence key so that 'Fix all occurrences' would work on all instances of the issue, e.g. "Use Assert.IsType<T>"